raddb/sites-available/ldap_sync

```diff
@@ -27,194 +102,391 @@
-		type = sync
+		transport = ldap
 
-		#  The LDAP server to connect to.
-		#
-		#  May be prefixed with:
-		#    - ldaps:// (LDAP over SSL)
-		#    - ldapi:// (LDAP over Unix socket)
-		#    - ldapc:// (Connectionless LDAP)
-		server = "localhost"
+		ldap {
+			#  The LDAP server to connect to.
+			#
+			#  May be prefixed with:
+			#    - ldaps:// (LDAP over SSL)
+			#    - ldapi:// (LDAP over Unix socket)
+			#    - ldapc:// (Connectionless LDAP)
+			server = "localhost"
+
+			#  Port to connect on, defaults to 389, will be ignored for LDAP URIs.
+#			port = 389
+
+			#  Administrator account for persistent search.
+			#  If using SASL + KRB5 these should be commented out.
+			identity = 'cn=admin,dc=example,dc=org'
+			password = mypass
+
+			options {
+				#
+				#  timeouts may need to be longer than for normal LDAP queries
+				#  if a refresh phase returns a lot of data.
+				#
+				res_timeout = 120
+				srv_timelimit = 60
+				idle = 60
+				probes = 3
+				interval = 3
+				reconnection_delay = 10
+			}
 
-		#  Port to connect on, defaults to 389, will be ignored for LDAP URIs.
-#		port = 389
+			#
+			#  SASL parameters to use for binding as the sync user.
+			#
+			sasl {
+				#  SASL mechanism
+#				mech = 'PLAIN'
 
-		#  Administrator account for persistent search.
-		#  If using SASL + KRB5 these should be commented out.
-#		identity = 'cn=admin,dc=example,dc=org'
-#		password = mypass
+				#  SASL authorisation identity to proxy.
+#				proxy = 'autz_id'
 
-		#  How long to wait after failing to initialise a persistent search
-		#  or after receiving malformed/unexpected messages from the LDAP server.
-#		sync_retry_interval = 5.0
+				#  SASL realm. Used for kerberos.
+#				realm = 'example.org'
+			}
 
-		#  How long to wait before reinitialising the connection if we become
-		#  disconnected from the LDAP directory.
-#		conn_retry_interval = 5.0
+			#
+			#  Persistent searches.
+			#
+			#  You can configure an unlimited (within reason, and any limitations
+			#  of the directory you are querying), number of syncs to retrieve
+			#  entries from the LDAP directory.
+			#
+			sync {
+				#  Where to start searching in the tree for entries
+				base_dn = "ou=people,dc=example,dc=com"
 
-		#
-		#  SASL parameters to use for binding as the sync user.
-		#
-		sasl {
-			#  SASL mechanism
-#			mech = 'PLAIN'
+				#
+				#  Only return entries matching this filter
+				#  Comment this out if all entries should be returned.
+				#
+				filter = "(objectClass=posixAccount)"
 
-			#  SASL authorisation identity to proxy.
-#			proxy = 'autz_id'
+				#  Search scope, may be 'base', 'one', 'sub' or 'children'
+				scope = 'sub'
 
-			#  SASL realm. Used for kerberos.
-#			realm = 'example.org'
-		}
+				#
+				#  Specify the list of attributes to return in entries,
+				#  one config item per LDAP attribute.
+				#
+				#  If no attr config items are specified, then all attributes
+				#  will be returned.
+				#
+				attrs = 'cn'
```

Comment:
These should probably be singular not plural, don't know who did that initially.